baseapp/baseapp.go

```diff
@@ -739,7 +739,7 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 		//
 		// Note: If the postHandler fails, we also revert the runMsgs state.
 		if app.postHandler != nil {
-			newCtx, err := app.postHandler(runMsgCtx, tx, mode == runTxModeSimulate)
+			newCtx, err := app.postHandler(runMsgCtx, tx, result, mode == runTxModeSimulate, err == nil)
```

Comment:
nit: `err == nil` is always false here, since we've evaluated that condition already just above.  https://github.com/cosmos/cosmos-sdk/blob/f4ef9d601608203114bd074f01b7a550adf94f63/baseapp/baseapp.go#L736-L741

Have you considered removing `success` as a function argument in `type PostHandler`?  There are no code paths were it could be true.

Reply:
The follow-up PR #13941 (issue #13942) changes this and removes the postHandler from the `if err == nil` block

Reply:
Okay.. let's just add a comment in code explaining this odd looking (to me) `err == nil` with maybe a link to the PR/issue?

Just in case the follow up somehow doesn't make it to main.  Stranger things have happened ;)